src/app/data-model-provider/ProviderMetadataTree.h

```diff
@@ -30,13 +31,33 @@ namespace chip {
 namespace app {
 namespace DataModel {
 
+struct NodeDataModelConfiguration
+{
+    uint32_t configurationVersion;
+    uint32_t dataModelVersion;
+    uint32_t specVersion;
+};
+
 /// Provides metadata information for a data model
 ///
 /// The data model can be viewed as a tree of endpoint/cluster/(attribute+commands+events)
 /// where each element can be iterated through independently.
 class ProviderMetadataTree
 {
 public:
+    // This sub-class provides the scoped where the configuration version get bumped.
+    // It is used to provide RAII for the ConfigurationVersion where the value is bumped when deconstructed
+    // and provides a stable interface for clients to perform the bump.
+    class ScopedConfigurationVersionUpdater
+    {
+    public:
+        ScopedConfigurationVersionUpdater(ProviderMetadataTree * parentTree) { mParentTree = parentTree; }
+        ~ScopedConfigurationVersionUpdater() { mParentTree->Internal_BumpNodeDataModelConfigurationVersion(); };
```

Comment:
So there's a concern.  Say you have a ProviderMetadataTree that multiplexes together a few other ProviderMetadataTree instances (e.g. one overall thing that then delegates to other instances for specific endpoints).

When the configuration version needs to get bumped, deep inside those multiplexed instances, they would need access to the thing doing the multiplexing to do the bumping, it seems like?

As in, it's not clear to me that ProviderMetadataTree is the right place to put this: there can be multiple of those around, but by definition ConfigurationVersion is a single thing....